### PR TITLE
Use new runconfig endpoint; add severities to check, skipcheck, hard-fail-on, soft-fail-on

### PR DESCRIPTION
**Pull request by mikeurbanski1**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

- Uses the new endpoint in the platform if you are using an API key
  - This prevents having separate API calls per integration feature (suppressions, custom policies, etc  - but fixes are still separate, because they occur later)
  - The integration features just consume what they need from the endpoint response
  - The API key endpoint returns check metadata (severities, BC ID mapping, guidelines, benchmarks, categories, prisma title, prisma severity, and more), custom policies, suppressions, repo configs)
  - The anonymous public endpoint that returns check ID mapping and guidelines still exists
- Consolidates check metadata logic into a new integration feature class (another step towards splitting out the behemoth of platform_integration)
- Adds severity logic to --check and --skip-check
  - If you put a severity for --check, it will run that severity and higher
  - If you use a severity for --skip-check, it will skip that severity and lower
  - You can combine --check and --skjp-check, in which case it will first apply the --check filter (remove anything not matching), and then substract anything matching --skip-check. It's covered in the docs
- Adds severity logic to --hard-fail and --soft-fail
  - Using a severity with soft-fail will soft-fail for that severity or lower
  - Using a severity with hard-fail will hard fail that severity or higher (effectively the same as the code review settings in the platform)
  - Specific check IDs take precedence over severities - so you can hard-fail HIGH (and above) but soft-fail CKV_123 (assume this is a high severity)
  - Covered more in docs
- Added logic to every runner to handle this
- Eliminated the min-cve-severity-flag and incorporated into the logic described above (this was decided in a thread a couple weeks ago)
- Eliminated the skip-suppressions, skip-custom-policies, and no-guide flags in favor of a single skip-download flag (but the other flags will trigger a warning)
- There is also the start of logic to fetch Prisma policy metadata (severity, title) - not used yet
- It also fetches other policy metadata - benchmarks, category
- This is a pre-req for the following future features:
  - Use Prisma policy metadata when a prisma API key is used
  - Add check / skip / etc logic for benchmarks, categories
  - Inherit more repo config settings to use in the run, mainly code review thresholds

### CODE REVIEW

File: checkov/common/bridgecrew/platform_integration.py
Comment (nimrodkor) [inline checkov/common/bridgecrew/platform_integration.py:194]: @mikeurbanski1 Garbage? If there's something to do here, please describe it 🙏

File: checkov/common/bridgecrew/platform_integration.py
Comment (nimrodkor) [inline checkov/common/bridgecrew/platform_integration.py:413]: @mikeurbanski1 This means we have an issue in prod, would upgrade the level here to info / warning

**Review by nimrodkor — COMMENTED** [review]
This has been waiting for far too long, really nice and complete. Can't have 0 comments though...

File: checkov/common/bridgecrew/integration_features/features/custom_policies_integration.py
Comment (mikeurbanski1) [reply checkov/common/bridgecrew/integration_features/features/custom_policies_integration.py:44]: Because I am not as smart as you (updated)

File: checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py
Comment (mikeurbanski1) [reply checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py:33]: No, because if the platform integration isn't configured, this won't run. But I think I can bump the level down to debug, as we always just get debug logs whenever there is unexpected behavior anyways.

File: checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py
Comment (mikeurbanski1) [reply checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py:40]: I struggled with this for awhile before giving up and doing this, but I will spend some more time on it

File: checkov/common/bridgecrew/integration_features/features/repo_config_integration.py
Comment (nimrodkor) [inline checkov/common/bridgecrew/integration_features/features/repo_config_integration.py:17]:
```suggestion
    def __init__(self):
        super().__init__(bc_integration, order=0)
```

**Review by nimrodkor — APPROVED** [review]
Would wait for another review, too many files. But looks good!

File: checkov/common/bridgecrew/integration_features/features/repo_config_integration.py
Comment (gruebel) [inline checkov/common/bridgecrew/integration_features/features/repo_config_integration.py:20]:
```suggestion
        self.code_review_skip_policies = set()
```
change this to a `set`

File: checkov/common/bridgecrew/integration_features/features/repo_config_integration.py
Comment (gruebel) [inline checkov/common/bridgecrew/integration_features/features/repo_config_integration.py:58]:
```suggestion
                        self.code_review_skip_policies.update(section['rule']['excludePolicies'])
```
then you can adjust it like this and safe the time to convert the `list` to a `set`

**Review by gruebel — APPROVED** [review]
nice job 🚀

**Review by schosterbarak — APPROVED** [review]

File: checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py
Comment (mikeurbanski1) [reply checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py:12]: Just a copy/paste error on my part

File: checkov/common/bridgecrew/integration_features/features/repo_config_integration.py
Comment (mikeurbanski1) [reply checkov/common/bridgecrew/integration_features/features/repo_config_integration.py:17]: I... can't remember why I changed this.

**Pull request merged**
